src/sensor/adxl345b/Adxl345b.c

```diff
@@ -244,328 +258,286 @@ adxl345bErrorCode_t adxl345bGetMultipleMeasurements(adxl345bSensorConfiguration_
+                                                ADXL345B_BITMASK_MEASURE);
     return errorCode;
 }
 
-static adxl345bErrorCode_t fetchDataFromFifo(adxl345bSensorConfiguration_t sensor,
-                                             uint8_t *dataResponseBuffer, uint8_t maxFifoRead) {
-    adxl345bErrorCode_t errorCode = ADXL345B_NO_ERROR;
-    for (uint8_t i = 0; i < maxFifoRead; i += 6) {
-        errorCode = adxl345bReadDataXYZ(sensor, dataResponseBuffer + i);
-
-        // sleep 5 μs to ensure data is ready
-        sleep_for_us(5);
-
-        if (errorCode != ADXL345B_NO_ERROR) {
-            PRINT_DEBUG("ReadDataXYZ failed");
-            return errorCode;
-        }
-    }
-    return errorCode;
-}
-
-adxl345bErrorCode_t adxl345bPerformSelfTest(adxl345bSensorConfiguration_t sensor, int *deltaX,
-                                            int *deltaY, int *deltaZ) {
+adxl345bErrorCode_t adxl345bPerformSelfTest(adxl345bSensorConfiguration_t sensor, int *averageX,
+                                            int *averageY, int *averageZ) {
     adxl345bErrorCode_t errorCode;
-
-    /* standard mode, 100Hz */
-    errorCode = adxl345bWriteConfigurationToSensor(sensor, ADXL345B_REGISTER_BW_RATE,
-                                                   ADXL345B_BW_RATE_LPM_12_point_5);
+    errorCode = adxl345bActivateMeasurementMode(sensor);
     if (errorCode != ADXL345B_NO_ERROR) {
         return errorCode;
     }
-    /* disable sleep/wakeup functions, start measurements */
-    errorCode = adxl345bWriteConfigurationToSensor(sensor, ADXL345B_REGISTER_POWER_CONTROL,
-                                                   ADXL345B_BW_RATE_25);
+    errorCode =
+        adxl345bWriteConfigurationToSensor(sensor, ADXL345B_REGISTER_BW_RATE, ADXL345B_BW_RATE_100);
     if (errorCode != ADXL345B_NO_ERROR) {
         return errorCode;
     }
-    /* right adjusted, 16G range */
     errorCode = adxl345bChangeMeasurementRange(sensor, ADXL345B_16G_RANGE);
     if (errorCode != ADXL345B_NO_ERROR) {
         return errorCode;
     }
+    errorCode = adxl345bInternalEnableSelftestForce(sensor);
+    if (errorCode != ADXL345B_NO_ERROR) {
+        return errorCode;
+    }
 
     sleep_for_ms(2);
 
-    /* collect 100 samples without force */
-    int samplesWithoutForce[100][3];
+    /* collect samples with force */
+    int numberOfSamples = 10;
+    int samplesWithForce[numberOfSamples][3];
     int counter = 0;
-    while (counter < 100) {
+    while (counter < numberOfSamples) {
         uint8_t interrupt_source;
         errorCode = adxl345bInternalReadDataFromSensor(sensor, ADXL345B_REGISTER_INTERRUPT_SOURCE,
                                                        &interrupt_source, 1);
         if (errorCode != ADXL345B_NO_ERROR) {
             return errorCode;
         }
 
-        if (interrupt_source & 0b10000000) {
+        if ((interrupt_source & ADXL345B_BITMASK_DATA_READY) == ADXL345B_BITMASK_DATA_READY) {
```

Comment:
Should be sufficient:
```suggestion
        if (interrupt_source & ADXL345B_BITMASK_DATA_READY) {
```